imports/api/eligibleClients/methods.js

```diff
@@ -23,6 +23,21 @@ Meteor.methods({
     return hc.api('house-matching-v2').getEligibleClientsPage(pageNumber, pageSize, sort, order);
   },
 
+  updateBonusScore(clientId, bonusScore) {
+    logger.info(`METHOD[${this.userId}]: updateScores(${Array.isArray(clientId)}, ${clientId})`); // eslint-disable-line max-len
+    check(clientId, Match.OneOf(String, [String])); // eslint-disable-line new-cap
+    if (!Roles.userIsInRole(this.userId, ClientsAccessRoles)) {
+      throw new Meteor.Error(403, 'Forbidden');
+    }
+
+    const hc = HmisClient.create(this.userId);
+    // get client details
+    const eligibleClient = hc.api('house-matching').getEligibleClient(clientId);
+    eligibleClient.bonusScore = bonusScore;
+    eligibleClient.totalScore = eligibleClient.bonusScore + eligibleClient.surveyScore / 1;
```

Comment:
What if you call this method twice in a row? shouldn't you remove the previous bonus from total first and then apply a new bonus?

Reply:
Bonus is recalculated on every call. Ignoring previous value